packages/components/src/InputFile/InputFile.tsx

```diff
@@ -139,6 +140,18 @@ interface InputFileProps {
    * Upload event handler. Triggered on upload completion.
    */
   onUploadComplete?(file: FileUpload): void;
+
+  /**
+   *  Upload event handler. Triggered on upload error.
+   */
+  onUploadError?(error: Error): void;
+
+  /**
+   * Pass a custom validator function that will be called when a file is dropped.
+   */
+  readonly validator?: <T extends File>(
```

Comment:
tempted to add an example validator to the docs so people can see how to implement their own

Reply:
I can add a section for validations